### PR TITLE
Feature/mob logic

### PR DESCRIPTION
**Pull request by HaniaDzierwa**
Feature/mob logic + small fixes opponents

### CODE REVIEW

File: Farm&Gun/Assets/Scripts/Entities/Opponents/Opponent.cs [hunk 1]
```diff
@@ -4,20 +4,18 @@
...
-    [SerializeField]
-    private float speed;
-
     [SerializeField]
     private float shotForce;
 
     [SerializeField]
     private NavMeshAgent agent;
 
+    private float buffAmount = 2;
```
Comment (kochamdziekana) [inline Farm&Gun/Assets/Scripts/Entities/Opponents/Opponent.cs:13]: If it isn't changed anywhere - make it a Constant

File: Farm&Gun/Assets/Scripts/Entities/Opponents/GenerateOpponent.cs [hunk 2]
```diff
@@ -6,22 +6,58 @@ public class GenerateOpponent : MonoBehaviour
...
     void InstantiateOponents()
     {
-        //TODO: if multiple add to list 
+        //WOLF
         GameObject wolfInstance = GameObject.Instantiate(wolf, this.transform.position, Quaternion.identity);
         wolfInstance.transform.rotation = transform.rotation;
         wolfs.Add(wolfInstance);
+
+        //FOX
+        //GameObject foxInstance = GameObject.Instantiate(fox, this.transform.position, Quaternion.identity);
```
Comment (kochamdziekana) [inline Farm&Gun/Assets/Scripts/Entities/Opponents/GenerateOpponent.cs:43]: Why is it commented? Remove if not used

File: Farm&Gun/Assets/Scripts/Entities/Opponents/GenerateOpponent.cs [hunk 3]
```diff
@@ -6,22 +6,58 @@ public class GenerateOpponent : MonoBehaviour
...
-    [SerializeField]
     private List<GameObject> wolfs, foxes;
 
+    [SerializeField]
+    private int amountWolf, amountFox;
+    
+    private void Awake()
+    {
+        wolfs = new ();
+        foxes = new ();
```
Comment (kochamdziekana) [inline Farm&Gun/Assets/Scripts/Entities/Opponents/GenerateOpponent.cs:17]: Won't this create errors?

**Review by kochamdziekana — CHANGES_REQUESTED** [review]

File: Farm&Gun/Assets/Scripts/Entities/Opponents/GenerateOpponent.cs (hunk 2 above)
Comment (HaniaDzierwa) [reply Farm&Gun/Assets/Scripts/Entities/Opponents/GenerateOpponent.cs:43]: Usefull for testing

File: Farm&Gun/Assets/Scripts/Entities/Opponents/GenerateOpponent.cs (hunk 3 above)
Comment (HaniaDzierwa) [reply Farm&Gun/Assets/Scripts/Entities/Opponents/GenerateOpponent.cs:17]: above is type definition, this is just initialization, type will be automaticly resolved

File: Farm&Gun/Assets/Scripts/Entities/Opponents/Opponent.cs (hunk 1 above)
Comment (HaniaDzierwa) [reply Farm&Gun/Assets/Scripts/Entities/Opponents/Opponent.cs]: fixed

**Pull request merged**